app/generators/checkbox_generator.rb

```diff
@@ -28,18 +28,20 @@ def answer_options(question)
     safe_join(body)
   end
 
-  def question_options(elem_id)
+  def question_options(question, elem_id)
+    previous_value = question[:previous_response]&.[](elem_id)
     {
       type: 'checkbox',
       id: elem_id,
       name: answer_name(elem_id),
-      value: true
+      value: true,
+      checked: previous_value.presence || nil
```

Comment:
Couldn't you also use the `decorate_with_previous_value` method here? with a yield and then you return like the `[:checked, previous_value.presence || nil]` here? Passing `elem_id` instead of `id`? It would make the code a bit more consistent and then the `question[:previous_response]&.[](...)` logic of how to retrieve the previous value stays isolated in one place (the `decorate_with_previous_value` method).

Reply:
oh yes, I implemented the decorator after this but forgot to replace it